packages/microcosm/src/domain.js

```diff
@@ -129,8 +137,22 @@ export class Domain<State: mixed = null> extends Agent {
     let { deserialize, data } = payload
     let { key } = this.options
 
-    let value = key in data ? data[key] : this.getInitialState()
+    console.assert(
+      meta.status === 'complete',
+      'Unable to reset or patch from incomplete action. This is an internal Microcosm error.'
+    )
+
+    console.assert(
+      data,
+      'Unable to reset or patch, no data provided. This is an internal Microcosm error.'
+    )
```

Comment:
These assertions were the best way to get test coverage on this case. Otherwise I have to dig into Microcosm internals in my tests, which is problematic for testing the production build.